crates/ark/src/modules/positron/r_data_explorer.R

```diff
@@ -327,13 +327,38 @@ export_selection <- function(x, format = c("csv", "tsv", "html"), include_header
 }
 
 write_delim <- function(x, delim, include_header) {
-    tmp <- tempfile()
-    defer(unlink(tmp))
+    path <- tempfile()
+    defer(unlink(path))
 
-    utils::write.table(x, tmp, sep = delim, row.names = FALSE, col.names = include_header, quote = FALSE, na = "")
-    # We use size - 1 because we don't want to read the last newline character
-    # that creates problems when pasting the content in spreadsheets
-    readChar(tmp, file.info(tmp)$size - 1L)
+    write_delim_impl(x, path, delim, include_header)
+
+    # We use `size - 1` because we don't want to read the last newline character
+    # as that creates problems when pasting the content in spreadsheets.
+    # `file.info()$size` reports the size in bytes, hence `useBytes = TRUE`.
+    readChar(path, file.info(path)$size - 1L, useBytes = TRUE)
+}
+
+write_delim_impl <- function(x, path, delim, include_header) {
+    # Scope the `con` lifetime to just this helper.
+    # We need to `close()` the connection before we try and get the
+    # `file.info()$size`.
+
+    # Must open in binary write mode, otherwise even though we set
+    # `eol = "\n"` on Windows it will still write `\r\n` due to the C
+    # level `vfprintf()` call.
+    con <- file(path, open = "wb")
+    defer(close(con))
```

Comment:
Bug found when adding Windows CI. Tests expect `\n` as the line ending

And the way we do `readChar(path, file.info(path)$size - 1L, useBytes = TRUE)` above expects 1 char for the trailing line (i.e. `\n` not `\r\n`)